ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -544,18 +677,38 @@ private SelectionKey keyForId(String id) {
     return this.keyMap.get(id);
   }
 
+  /**
+   * Handle read/write IOException
+   */
+  private void handleReadWriteIOException(IOException e, SelectionKey key) {
+    String socketDescription = socketDescription(channel(key));
+    if (e instanceof EOFException || e instanceof ConnectException) {
+      metrics.selectorDisconnectedErrorCount.inc();
+      logger.error("Connection {} disconnected", socketDescription, e);
+    } else {
+      metrics.selectorIOErrorCount.inc();
+      logger.warn("Error in I/O with connection to {}", socketDescription, e);
```

Comment:
why here we use `warn` instead of `error` level?

Reply:
It was `warn` before. It may cause by network issue and is not critical to us.